griptape/loaders/image_loader.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+import os
+from io import BytesIO
+from pathlib import Path
+from attr import define
+from griptape import utils
+from griptape.artifacts import ImageArtifact, ErrorArtifact
+from griptape.loaders import BaseLoader
+from PIL import Image
+
+
+@define
+class ImageLoader(BaseLoader):
+    def load(self, path: str | Path) -> ImageArtifact | ErrorArtifact:
+        return self.file_to_artifact(path)
+
+    def load_collection(self, paths: list[str | Path]) -> dict[str, ImageArtifact | ErrorArtifact]:
+        return utils.execute_futures_dict(
+            {utils.str_to_hash(str(path)): self.futures_executor.submit(self.file_to_artifact, path) for path in paths}
+        )
+
+    def file_to_artifact(self, path: str | Path) -> ImageArtifact | ErrorArtifact:
+        file_name = os.path.basename(path)
+        dir_name = os.path.dirname(path)
+
+        try:
+            image = Image.open(path)
+
+            # Ensure image is in png format.
+            byte_stream = BytesIO()
+            image.save(byte_stream, format="png")
```

Comment:
All the models we support exclusively handle PNG, so the motivation of this is to provide some end-user flexibility to ensure the ImageArtifact bytes are formatted correctly. I've moved the format to a field for additional flexibility (because why limit ourselves to PNG), added a check to avoid redundant reformatting if the incoming image already matches the target format, and added a test to prove the functionality.